core/modules/search/search.admin.inc

```diff
@@ -79,6 +78,28 @@ function search_admin_settings($form, &$form_state) {
     '#options' => $module_options,
     '#description' => t('Only one type of item will be searched by default. This selection will determine what appears at http://www.example.com/search.')
   );
+  $form['#validate'][] = 'search_admin_settings_validate';
+  $form['#submit'][] = 'search_admin_settings_submit';
+
+  // Per module search ranking settings.
+  foreach ($config->get('search_active_modules') as $module) {
+    $factors = module_invoke($module, 'search_admin');
+    // Only render this section if there are any ranking factors.
+    if (is_array($factors)) {
+      $factors += $factors;
+
+      // Common fieldset for all influence factors implemented by various
+      // modules via hook_search_admin().
+      $form['content_ranking'] = array(
+        '#type' => 'fieldset',
+        '#title' => t('Content ranking'),
+        '#description' => t('Influence is a numeric multiplier used in ordering search results. A higher number means the corresponding factor has more influence on search results; zero means the factor is ignored. Changing these numbers does not require the search index to be rebuilt. Changes take effect immediately.'),
+        '#theme' => 'node_search_admin',
+      );
+      $form['content_ranking']['factors'] = $factors;
```

Comment:
If I'm reading this right, every module that implements `hook_search_admin()` will then re-add the same form element of `$form['content_ranking']`, replacing the previous one. Although I think this gets the "right" result, it seems like this should be moved outside of the `foreach()` loop rather than setting the same form element each time.

Reply:
The issue was closed as outdated.